src/ontology/mondo-ingest.Makefile

```diff
@@ -594,9 +598,10 @@ update-externally-managed-content: external-content-nord external-content-ordo
 mapped-deprecated-terms: mapped-deprecated-terms-docs
 
 .PHONY: mapped-deprecated-terms-docs
-mapped-deprecated-terms-docs: ../../docs/reports/mapped_deprecated.md
+mapped-deprecated-terms-docs: $(DOCS_DIR)/reports/mapped_deprecated.md
 
-../../docs/reports/mapped_deprecated.md: mapped-deprecated-terms-artefacts
+# todo: ideally unmapped_docs.py would also take file path(s) as input args. Currently, running `docs/reports/unmapped_%.md` for any source will trigger this for all sources.
+$(DOCS_DIR)/reports/mapped_deprecated.md $(DOCS_DIR)/reports/mapped_deprecated_%.md: $(foreach n,$(ALL_COMPONENT_IDS), $(REPORTDIR)/$(n)_mapped_deprecated_terms.robot.template.tsv)
```

Comment:
## Update to `$(DOCS_DIR)/reports/mapped_deprecated.md $(DOCS_DIR)/reports/mapped_deprecated_%.md:`
- Previously did not have `$(DOCS_DIR)/reports/mapped_deprecated_%.md` as a target on the goal
- Previously, the prereq was a `.PHONY` goal, which means it'll run every time, even if the inputs don't need to be updated.

Also added a todo:
> ideally `deprecated_in_mondo.py` would also take file path(s) as input args. Currently, running `docs/reports/mapped_deprecated_%.md` for any source will trigger this for all sources.

Reply:
Its a weird way of doing dependencies, but not your fault. I would probably prefer this:

```suggestion
$(DOCS_DIR)/reports/mapped_deprecated.md $(foreach n,$(ALL_COMPONENT_IDS), $(DOCS_DIR)/reports/mapped_deprecated_$(n).md): $(foreach n,$(ALL_COMPONENT_IDS), $(REPORTDIR)/$(n)_mapped_deprecated_terms.robot.template.tsv)
```

But you will have to think it through. Maybe it is exactly the same, but I am worried the goal is executed multiple times with the current solution. Maybe I am wrong.

Reply:
@matentzn Interesting pattern! I am willing to make the change, but only after you consider this.

I have started to use `--dry-run` to test the differences between such changes, troubleshoot, etc. It gives you an instantaneous look at everything that would have run / all of the `make` output.

I ran `make ../../docs/reports/mapped_deprecated_doid.md --dry-run` and `make ../../docs/reports/mapped_deprecated.md --dry-run`, 1 time each for each of these variations above in the commit suggestion box; once before the change, and once after. 4 runs total.

I got the same exact result all 4 times; no diff.
<details><summary>Resulting log</summary>
<p>

```
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I http://purl.obolibrary.org/obo/doid.owl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-doid.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-doid.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-doid.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-doid.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-doid.tsv
(head -n 1 reports/mirror_signature-doid.tsv && tail -n +2 reports/mirror_signature-doid.tsv | sort) > reports/mirror_signature-doid.tsv-temp
mv reports/mirror_signature-doid.tsv-temp reports/mirror_signature-doid.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-doid.owl.owl" -q "../sparql/doid-relevant-signature.sparql" tmp/doid_relevant_signature.txt; fi
if [ true  = true ]; then robot --catalog catalog-v001.xml remove -i tmp/component-download-doid.owl.owl --select imports \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        remove -T tmp/doid_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        query \
            --update ../sparql/fix_omimps.ru \
            --update ../sparql/fix-labels-with-brackets.ru \
            --update ../sparql/rm_xref_by_prefix.ru \
            --update ../sparql/fix_make_omim_exact.ru \
        remove -T config/properties.txt --select complement --select properties --trim true \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/doid.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/doid.owl -o components/doid.owl; fi
robot --catalog catalog-v001.xml query -i components/doid.owl --query ../sparql/classes.sparql reports/component_signature-doid.tsv
(head -n 1 reports/component_signature-doid.tsv && tail -n +2 reports/component_signature-doid.tsv | sort) > reports/component_signature-doid.tsv-temp
mv reports/component_signature-doid.tsv-temp reports/component_signature-doid.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/doid_exclusions.tsv \
    --onto-path tmp/component-download-doid.owl.owl \
    --mirror-signature-path reports/mirror_signature-doid.tsv \
    --component-signature-path reports/component_signature-doid.tsv \
    --config-path metadata/doid.yml \
    --outpath-txt reports/doid_term_exclusions.txt \
    --outpath-robot-template-tsv reports/doid_exclusion_reasons.robot.template.tsv
cp tmp/mondo/src/ontology/mappings/mondo.sssom.tsv tmp/mondo.sssom.tsv
rm -f components/doid.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/doid-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/doid.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/doid-relation-graph.tsv.gz
test -f components/doid.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/doid_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/doid.yml \
    --db-path components/doid.db \
    --outpath-simple reports/doid_unmapped_terms.tsv \
    --outpath-full reports/doid_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/doid_mapping_status.tsv \
    --outpath reports/doid_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://github.com/monarch-initiative/gard/releases/latest/download/gard.owl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-gard.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-gard.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-gard.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-gard.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-gard.tsv
(head -n 1 reports/mirror_signature-gard.tsv && tail -n +2 reports/mirror_signature-gard.tsv | sort) > reports/mirror_signature-gard.tsv-temp
mv reports/mirror_signature-gard.tsv-temp reports/mirror_signature-gard.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-gard.owl.owl" -q "../sparql/gard-relevant-signature.sparql" tmp/gard_relevant_signature.txt; fi
if [ true  = true ]; then robot --catalog catalog-v001.xml remove -i tmp/component-download-gard.owl.owl --select imports \
        remove -T tmp/gard_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/gard.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/gard.owl -o components/gard.owl; fi
robot --catalog catalog-v001.xml query -i components/gard.owl --query ../sparql/classes.sparql reports/component_signature-gard.tsv
(head -n 1 reports/component_signature-gard.tsv && tail -n +2 reports/component_signature-gard.tsv | sort) > reports/component_signature-gard.tsv-temp
mv reports/component_signature-gard.tsv-temp reports/component_signature-gard.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/gard_exclusions.tsv \
    --onto-path tmp/component-download-gard.owl.owl \
    --mirror-signature-path reports/mirror_signature-gard.tsv \
    --component-signature-path reports/component_signature-gard.tsv \
    --config-path metadata/gard.yml \
    --outpath-txt reports/gard_term_exclusions.txt \
    --outpath-robot-template-tsv reports/gard_exclusion_reasons.robot.template.tsv
rm -f components/gard.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/gard-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/gard.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/gard-relation-graph.tsv.gz
test -f components/gard.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/gard_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/gard.yml \
    --db-path components/gard.db \
    --outpath-simple reports/gard_unmapped_terms.tsv \
    --outpath-full reports/gard_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/gard_mapping_status.tsv \
    --outpath reports/gard_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ]; then wget "https://data.bioontology.org/ontologies/ICD10CM/submissions/22/download?apikey=8b5b7825-538d-40e0-9e9e-5ab9274a9aeb" -O tmp/icd10cm.tmp.owl && robot --catalog catalog-v001.xml remove -i tmp/icd10cm.tmp.owl --select imports \
        remove -T config/remove_properties.txt \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-icd10cm.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-icd10cm.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-icd10cm.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-icd10cm.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-icd10cm.tsv
(head -n 1 reports/mirror_signature-icd10cm.tsv && tail -n +2 reports/mirror_signature-icd10cm.tsv | sort) > reports/mirror_signature-icd10cm.tsv-temp
mv reports/mirror_signature-icd10cm.tsv-temp reports/mirror_signature-icd10cm.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-icd10cm.owl.owl" -q "../sparql/icd10cm-relevant-signature.sparql" tmp/icd10cm_relevant_signature.txt; fi
if [ true  = true ]; then robot --catalog catalog-v001.xml merge -i tmp/component-download-icd10cm.owl.owl \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        remove -T tmp/icd10cm_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        remove -T tmp/icd10cm_relevant_signature.txt --select individuals \
        query \
            --update ../sparql/fix_omimps.ru \
            --update ../sparql/fix-labels-with-brackets.ru \
        remove -T config/properties.txt --select complement --select properties --trim true \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/icd10cm.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/icd10cm.owl -o components/icd10cm.owl; fi
robot --catalog catalog-v001.xml query -i components/icd10cm.owl --query ../sparql/classes.sparql reports/component_signature-icd10cm.tsv
(head -n 1 reports/component_signature-icd10cm.tsv && tail -n +2 reports/component_signature-icd10cm.tsv | sort) > reports/component_signature-icd10cm.tsv-temp
mv reports/component_signature-icd10cm.tsv-temp reports/component_signature-icd10cm.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/icd10cm_exclusions.tsv \
    --onto-path tmp/component-download-icd10cm.owl.owl \
    --mirror-signature-path reports/mirror_signature-icd10cm.tsv \
    --component-signature-path reports/component_signature-icd10cm.tsv \
    --config-path metadata/icd10cm.yml \
    --outpath-txt reports/icd10cm_term_exclusions.txt \
    --outpath-robot-template-tsv reports/icd10cm_exclusion_reasons.robot.template.tsv
rm -f components/icd10cm.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/icd10cm-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/icd10cm.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/icd10cm-relation-graph.tsv.gz
test -f components/icd10cm.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/icd10cm_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/icd10cm.yml \
    --db-path components/icd10cm.db \
    --outpath-simple reports/icd10cm_unmapped_terms.tsv \
    --outpath-full reports/icd10cm_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/icd10cm_mapping_status.tsv \
    --outpath reports/icd10cm_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://github.com/monarch-initiative/icd10who/releases/latest/download/icd10who.ttl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-icd10who.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-icd10who.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-icd10who.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-icd10who.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-icd10who.tsv
(head -n 1 reports/mirror_signature-icd10who.tsv && tail -n +2 reports/mirror_signature-icd10who.tsv | sort) > reports/mirror_signature-icd10who.tsv-temp
mv reports/mirror_signature-icd10who.tsv-temp reports/mirror_signature-icd10who.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-icd10who.owl.owl" -q "../sparql/icd10who-relevant-signature.sparql" tmp/icd10who_relevant_signature.txt; fi
if [ true  = true ] ; then robot --catalog catalog-v001.xml remove -i tmp/component-download-icd10who.owl.owl --select imports \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        remove -T tmp/icd10who_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        remove -T tmp/icd10who_relevant_signature.txt --select individuals \
        query \
            --update ../sparql/fix_omimps.ru \
            --update ../sparql/fix-labels-with-brackets.ru \
        remove -T config/properties.txt --select complement --select properties --trim true \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/icd10who.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/icd10who.owl -o components/icd10who.owl; fi
robot --catalog catalog-v001.xml query -i components/icd10who.owl --query ../sparql/classes.sparql reports/component_signature-icd10who.tsv
(head -n 1 reports/component_signature-icd10who.tsv && tail -n +2 reports/component_signature-icd10who.tsv | sort) > reports/component_signature-icd10who.tsv-temp
mv reports/component_signature-icd10who.tsv-temp reports/component_signature-icd10who.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/icd10who_exclusions.tsv \
    --onto-path tmp/component-download-icd10who.owl.owl \
    --mirror-signature-path reports/mirror_signature-icd10who.tsv \
    --component-signature-path reports/component_signature-icd10who.tsv \
    --config-path metadata/icd10who.yml \
    --outpath-txt reports/icd10who_term_exclusions.txt \
    --outpath-robot-template-tsv reports/icd10who_exclusion_reasons.robot.template.tsv
rm -f components/icd10who.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/icd10who-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/icd10who.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/icd10who-relation-graph.tsv.gz
test -f components/icd10who.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/icd10who_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/icd10who.yml \
    --db-path components/icd10who.db \
    --outpath-simple reports/icd10who_unmapped_terms.tsv \
    --outpath-full reports/icd10who_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/icd10who_mapping_status.tsv \
    --outpath reports/icd10who_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://github.com/monarch-initiative/icd11/releases/latest/download/icd11foundation.owl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-icd11foundation.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-icd11foundation.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-icd11foundation.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-icd11foundation.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-icd11foundation.tsv
(head -n 1 reports/mirror_signature-icd11foundation.tsv && tail -n +2 reports/mirror_signature-icd11foundation.tsv | sort) > reports/mirror_signature-icd11foundation.tsv-temp
mv reports/mirror_signature-icd11foundation.tsv-temp reports/mirror_signature-icd11foundation.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-icd11foundation.owl.owl" -q "../sparql/icd11foundation-relevant-signature.sparql" tmp/icd11foundation_relevant_signature.txt; fi
if [ true  = true ] ; then robot --catalog catalog-v001.xml remove -i tmp/component-download-icd11foundation.owl.owl --select imports \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        rename --mappings config/icd11foundation-property-map.sssom.tsv \
        remove -T tmp/icd11foundation_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        remove -T tmp/icd11foundation_relevant_signature.txt --select individuals \
        query \
            --update ../sparql/fix-labels-with-brackets.ru \
        remove -T config/properties.txt --select complement --select properties --trim true \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/icd11foundation.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/icd11foundation.owl -o components/icd11foundation.owl; fi
robot --catalog catalog-v001.xml query -i components/icd11foundation.owl --query ../sparql/classes.sparql reports/component_signature-icd11foundation.tsv
(head -n 1 reports/component_signature-icd11foundation.tsv && tail -n +2 reports/component_signature-icd11foundation.tsv | sort) > reports/component_signature-icd11foundation.tsv-temp
mv reports/component_signature-icd11foundation.tsv-temp reports/component_signature-icd11foundation.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/icd11foundation_exclusions.tsv \
    --onto-path tmp/component-download-icd11foundation.owl.owl \
    --mirror-signature-path reports/mirror_signature-icd11foundation.tsv \
    --component-signature-path reports/component_signature-icd11foundation.tsv \
    --config-path metadata/icd11foundation.yml \
    --outpath-txt reports/icd11foundation_term_exclusions.txt \
    --outpath-robot-template-tsv reports/icd11foundation_exclusion_reasons.robot.template.tsv
rm -f components/icd11foundation.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/icd11foundation-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/icd11foundation.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/icd11foundation-relation-graph.tsv.gz
test -f components/icd11foundation.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/icd11foundation_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/icd11foundation.yml \
    --db-path components/icd11foundation.db \
    --outpath-simple reports/icd11foundation_unmapped_terms.tsv \
    --outpath-full reports/icd11foundation_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/icd11foundation_mapping_status.tsv \
    --outpath reports/icd11foundation_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I http://purl.obolibrary.org/obo/ncit.owl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-ncit.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-ncit.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-ncit.owl.owl; fi
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-ncit.owl.owl" -q "../sparql/ncit-relevant-signature.sparql" tmp/ncit_relevant_signature.txt; fi
if [ false = false ] && [ true  = true ]; then robot --catalog catalog-v001.xml remove -i tmp/component-download-ncit.owl.owl --select imports \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        query --update ../sparql/rm_xref_by_prefix.ru \
        remove -T tmp/ncit_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        remove -T config/properties.txt --select complement --select properties --trim true \
        remove --term "http://purl.obolibrary.org/obo/NCIT_C179199" --axioms "equivalent" \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/ncit.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/ncit.owl -o components/ncit.owl; fi
rm -f components/ncit.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/ncit-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/ncit.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/ncit-relation-graph.tsv.gz
test -f components/ncit.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/mirror_signature_via_oak.py --db-path components/ncit.db --onto-config-path metadata/ncit.yml --outpath reports/mirror_signature-ncit.tsv
robot --catalog catalog-v001.xml query -i components/ncit.owl --query ../sparql/classes.sparql reports/component_signature-ncit.tsv
(head -n 1 reports/component_signature-ncit.tsv && tail -n +2 reports/component_signature-ncit.tsv | sort) > reports/component_signature-ncit.tsv-temp
mv reports/component_signature-ncit.tsv-temp reports/component_signature-ncit.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/ncit_exclusions.tsv \
    --onto-path tmp/component-download-ncit.owl.owl \
    --mirror-signature-path reports/mirror_signature-ncit.tsv \
    --component-signature-path reports/component_signature-ncit.tsv \
    --config-path metadata/ncit.yml \
    --outpath-txt reports/ncit_term_exclusions.txt \
    --outpath-robot-template-tsv reports/ncit_exclusion_reasons.robot.template.tsv
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/ncit_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/ncit.yml \
    --db-path components/ncit.db \
    --outpath-simple reports/ncit_unmapped_terms.tsv \
    --outpath-full reports/ncit_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/ncit_mapping_status.tsv \
    --outpath reports/ncit_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://github.com/monarch-initiative/omim/releases/latest/download/omim.owl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-omim.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-omim.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-omim.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-omim.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-omim.tsv
(head -n 1 reports/mirror_signature-omim.tsv && tail -n +2 reports/mirror_signature-omim.tsv | sort) > reports/mirror_signature-omim.tsv-temp
mv reports/mirror_signature-omim.tsv-temp reports/mirror_signature-omim.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i "tmp/component-download-omim.owl.owl" -q "../sparql/omim-relevant-signature.sparql" tmp/omim_relevant_signature.txt; fi
if [ true  = true ]; then robot --catalog catalog-v001.xml remove -i tmp/component-download-omim.owl.owl --select imports \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        remove -T tmp/omim_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        remove -T config/remove.txt --axioms equivalent \
        query \
            --update ../sparql/fix_omimps.ru \
            --update ../sparql/fix-labels-with-brackets.ru \
            --update ../sparql/fix_illegal_punning_omim.ru \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/omim.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/omim.owl -o components/omim.owl; fi
robot --catalog catalog-v001.xml query -i components/omim.owl --query ../sparql/classes.sparql reports/component_signature-omim.tsv
(head -n 1 reports/component_signature-omim.tsv && tail -n +2 reports/component_signature-omim.tsv | sort) > reports/component_signature-omim.tsv-temp
mv reports/component_signature-omim.tsv-temp reports/component_signature-omim.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/omim_exclusions.tsv \
    --onto-path tmp/component-download-omim.owl.owl \
    --mirror-signature-path reports/mirror_signature-omim.tsv \
    --component-signature-path reports/component_signature-omim.tsv \
    --config-path metadata/omim.yml \
    --outpath-txt reports/omim_term_exclusions.txt \
    --outpath-robot-template-tsv reports/omim_exclusion_reasons.robot.template.tsv
rm -f components/omim.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/omim-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/omim.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/omim-relation-graph.tsv.gz
test -f components/omim.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/omim_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/omim.yml \
    --db-path components/omim.db \
    --outpath-simple reports/omim_unmapped_terms.tsv \
    --outpath-full reports/omim_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/omim_mapping_status.tsv \
    --outpath reports/omim_mapped_deprecated_terms.robot.template.tsv
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I http://www.orphadata.org/data/ORDO/ordo_orphanet.owl \
    annotate --ontology-iri http://purl.obolibrary.org/obo/mondo-ingest/component-download-ordo.owl annotate -V http://purl.obolibrary.org/obo/mondo-ingest/releases/2024-06-03/component-download-ordo.owl --annotation owl:versionInfo 2024-06-03 -o tmp/component-download-ordo.owl.owl; fi
robot --catalog catalog-v001.xml query -i tmp/component-download-ordo.owl.owl --query ../sparql/classes.sparql reports/mirror_signature-ordo.tsv
(head -n 1 reports/mirror_signature-ordo.tsv && tail -n +2 reports/mirror_signature-ordo.tsv | sort) > reports/mirror_signature-ordo.tsv-temp
mv reports/mirror_signature-ordo.tsv-temp reports/mirror_signature-ordo.tsv
if [ true  = true ]; then robot --catalog catalog-v001.xml query -i tmp/component-download-ordo.owl.owl --update ../sparql/ordo-construct-subclass-from-part-of.ru \
        query -q "../sparql/ordo-relevant-signature.sparql" tmp/ordo_relevant_signature.txt; fi
if [ true  = true ]; then robot --catalog catalog-v001.xml remove -i tmp/component-download-ordo.owl.owl --select imports \
        merge \
        rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
        query \
            --update ../sparql/fix_partof.ru \
            --update ../sparql/fix_deprecated.ru \
            --update ../sparql/fix_complex_reification_ordo.ru \
            --update ../sparql/fix_xref_prefixes.ru \
            --update ../sparql/fix-labels-with-brackets.ru \
            --update ../sparql/ordo-construct-subclass-from-part-of.ru \
            --update ../sparql/ordo-construct-subsets.ru \
            --update ../sparql/ordo-construct-d2g.ru \
            --update ../sparql/ordo-replace-annotation-based-mappings.ru \
        remove -T tmp/ordo_relevant_signature.txt --select complement --select "classes individuals" --trim false \
        remove -T config/properties.txt --select complement --select properties --trim true \
        annotate --ontology-iri http://purl.obolibrary.org/obo/mondo/sources/ordo.owl --version-iri http://purl.obolibrary.org/obo/mondo/sources/2024-06-03/ordo.owl -o components/ordo.owl; fi
robot --catalog catalog-v001.xml query -i components/ordo.owl --query ../sparql/classes.sparql reports/component_signature-ordo.tsv
(head -n 1 reports/component_signature-ordo.tsv && tail -n +2 reports/component_signature-ordo.tsv | sort) > reports/component_signature-ordo.tsv-temp
mv reports/component_signature-ordo.tsv-temp reports/component_signature-ordo.tsv
python3 ../scripts/exclusion_table_creation.py \
    --select-intensional-exclusions-path config/ordo_exclusions.tsv \
    --onto-path tmp/component-download-ordo.owl.owl \
    --mirror-signature-path reports/mirror_signature-ordo.tsv \
    --component-signature-path reports/component_signature-ordo.tsv \
    --config-path metadata/ordo.yml \
    --outpath-txt reports/ordo_term_exclusions.txt \
    --outpath-robot-template-tsv reports/ordo_exclusion_reasons.robot.template.tsv
rm -f components/ordo.db
rm -f .template.db
rm -f .template.db.tmp
rm -f components/ordo-relation-graph.tsv.gz
RUST_BACKTRACE=full semsql make components/ordo.db -P config/prefixes.csv
rm -f .template.db
rm -f .template.db.tmp
rm -f components/ordo-relation-graph.tsv.gz
test -f components/ordo.db || (echo "Error: File not found!" && exit 1)
python3 ../scripts/unmapped_tables.py \
    --exclusions-path reports/ordo_term_exclusions.txt \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --onto-config-path metadata/ordo.yml \
    --db-path components/ordo.db \
    --outpath-simple reports/ordo_unmapped_terms.tsv \
    --outpath-full reports/ordo_mapping_status.tsv
python3 ../scripts/deprecated_in_mondo.py \
    --mondo-mappings-path tmp/mondo.sssom.tsv \
    --mapping-status-path reports/ordo_mapping_status.tsv \
    --outpath reports/ordo_mapped_deprecated_terms.robot.template.tsv
python3 ../scripts/deprecated_in_mondo.py --docs
```

</p>
</details> 

Based on this, do you still want the change? Or, can you think of a different `--dry-run` that I can do to see if there is really any difference in these 2 variations?

Reply:
No opinion, you choose!

Reply:
I'll keep as-is then, as that is the standard pattern, though your new pattern is interesting! I hadn't thought of doing it that way.